src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -232,12 +231,35 @@ namespace cryptonote
       }
       summary_inputs_money += src_entr.amount;
 
+      // disposable addressing scheme
+      boost::optional<keypair> disposable_key;
+      {
+        char data[HASH_SIZE + 8];
+        memcpy(data, &sender_account_keys.m_view_secret_key, HASH_SIZE);
+        memcpy(data + HASH_SIZE, &src_entr.payment_id8, 8);
+        crypto::secret_key  disposable_seckey;
+        crypto::hash_to_scalar(data, HASH_SIZE + 8, disposable_seckey);
+        // check if the following equation holds: (H(viewkey, pID) - pID) mod 256 == 0
+        if (src_entr.payment_id8.data[0] == disposable_seckey.data[0])
```

Comment:
Thanks, I was missing that point.

I chose the number 8 rather arbitrarily, and the potential (theoretical?) "leakage" you described would decrease as this number is lowered, at the cost of increased wallet scanning time. The extreme in the other end would be to use all the 64 bits, which still leaves the search space of 256-64=192 bits for the secret scalar `m~`. We can try to come up with a better choice of this number hitting the "right" balance, but it seems difficult to me to assess the potential privacy leak quantitatively (or to judge whether it's a real problem or not in the first place).